src/cascadia/TerminalSettingsModel/FolderEntry.cpp

```diff
@@ -43,6 +49,72 @@ winrt::com_ptr<NewTabMenuEntry> FolderEntry::FromJson(const Json::Value& json)
     JsonUtils::GetValueForKey(json, NameKey, entry->_Name);
     JsonUtils::GetValueForKey(json, IconKey, entry->_Icon);
     JsonUtils::GetValueForKey(json, EntriesKey, entry->_Entries);
+    JsonUtils::GetValueForKey(json, InliningKey, entry->_Inlining);
+    JsonUtils::GetValueForKey(json, AllowEmptyKey, entry->_AllowEmpty);
 
     return entry;
 }
+
+// A FolderEntry should only expose the entries to actually render to WinRT,
+// to keep the logic for collapsing/expanding more centralised.
+using NTMEModel = winrt::Microsoft::Terminal::Settings::Model::NewTabMenuEntry;
+IVector<NTMEModel> FolderEntry::Entries() const

```

Comment:
nit: I might recommend calling this `ResolvedEntries` to make clear that it's gonna do some logic on the things inside it, unless `Entries` is highly conserved among all members of the `NewTabMenuEntry` tree of classes